server/channels/api4/websocket.go

```diff
@@ -30,6 +30,10 @@ func connectWebSocket(c *Context, w http.ResponseWriter, r *http.Request) {
 		CheckOrigin:     c.App.OriginChecker(),
 	}
 
+	if !upgrader.CheckOrigin(r) {
+		c.Logger.Warn("URL Blocked Because of CORS", mlog.String("Blocked Origin", r.Header.Get("Origin")))
+	}
+
```

Comment:
The change is fine, but the downside of how it's implemented is that `upgrader.CheckOrigin` gets called twice. I wonder if we can get the log message into the checker method. Is it an option to pass a logger to `c.App.OriginChecker` and log the warning in there?